Doc/library/importlib.rst

```diff
@@ -1584,20 +1584,30 @@ Note that if ``name`` is a submodule (contains a dot),
 Importing a source file directly
 ''''''''''''''''''''''''''''''''
 
-To import a Python source file directly, use the following recipe::
+This recipe should be used with caution: it is an approximation of an import statement where the file path is specified directly, rather than :data:`sys.path` being searched.
+Alternatives should first be considered, such as modifying :data:`sys.path` when a proper module is required, or using :func:`runpy.run_path` when the global namespace resulting from running a Python file is appropriate.
```

Comment:
```suggestion
This recipe should be used with caution: it is an approximation of an import
statement where the file path is specified directly, rather than
:data:`sys.path` being searched. Alternatives should first be considered first,
such as modifying :data:`sys.path` when a proper module is required, or using
:func:`runpy.run_path` when the global namespace resulting from running a Python
file is appropriate.
```